pkg/federate/kubefed/resource_test.go

```diff
@@ -0,0 +1,308 @@
+package kubefed
+
+import (
+	"context"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	ctlutil "sigs.k8s.io/kubefed/pkg/controller/util"
+	kubefedopt "sigs.k8s.io/kubefed/pkg/kubefedctl/options"
+)
+
+var _ = Describe("Federator", func() {
+	Describe("function Distribute", testDistribute)
+	Describe("helper function createFederatedResource", testCreateFederatedResource)
+})
+
+func testCreateFederatedResource() {
+	var (
+		resourceTyped      *corev1.Pod
+		expectedTyped      *unstructured.Unstructured
+		resource, expected runtime.Object
+		f                  *federator
+		err                error
+		initObjs           []runtime.Object
+		scheme             *runtime.Scheme
+	)
+
+	BeforeEach(func() {
+		resourceTyped = newPod("test-pod")
+		resource = resourceTyped
+		scheme = runtime.NewScheme()
+	})
+
+	JustBeforeEach(func() {
+		f = newFederatorWithScheme(scheme, initObjs...)
+		expectedTyped, err = f.createFederatedResource(resource)
+		expected = expectedTyped
+	})
+
+	When("the resource type scheme has NOT been added to the type registry", func() {
+		When("failing to convert to Unstructured", func() {
+			It("should throw an error", func() {
+				Expect(err).To(MatchError(HavePrefix(errorUnstructuredConversion, "")))
+			})
+
+			It("should return a null expected", func() {
+				Expect(expectedTyped).To(BeNil())
+			})
+		})
+	})
+
+	When("the resource type scheme has been added to the type registry", func() {
+		BeforeEach(func() {
+			corev1.AddToScheme(scheme)
```

Comment:
(no action required, just different opinion)
Again I think it's more clear to the reader where does expected/err come from exactly without the need to scroll.

IMO it's better to have a bit more code, but make the tests easier to read/explicit, in contrast to more compact code.
Of course, it's a trade off, and just my opinion, not sure what others think.